src/AWS/Decode.elm

```diff
@@ -83,3 +81,45 @@ optional fields decoder =
                             Err err ->
                                 JD.fail (toString err)
             )
+
+
+tryFields : List String -> JD.Decoder a -> JD.Decoder (Maybe JD.Value)
+tryFields fields decoder =
+    fields
+        |> List.map (\field -> JD.field field JD.value)
+        |> JD.oneOf
+        |> JD.maybe
+
+
+
+-- dict and helpers
+
+
+dict : JD.Decoder a -> JD.Decoder (Dict String a)
+dict valueDecoder =
+    [ JD.dict valueDecoder
+    , dictAsList valueDecoder
+    ]
+        |> JD.oneOf
```

Comment:
Using `oneOf` here to cover multiple possible representations of key values pairs in json. The default dict decoder just assumes a json object. The `dictAsList` is what `getQueueAttributes` actually uses